frontend/src/components/activitiesPanel.js

```diff
@@ -6,16 +6,21 @@ import { UserContext } from "../utils/userContext";
 import Activity from "./activity";
 import * as activityService from "../services/activityService";
 
-const ActivitiesPanel = ({ otherUserID }) => {
+const ActivitiesPanel = ({ otherUserID, all }) => {
   const [activities, setActivities] = useState();
   const { user } = useContext(UserContext);
 
   useEffect(() => {
     const fetchData = async () => {
-      const { data } = await activityService.getActivities(
-        otherUserID ? otherUserID : user.id
-      );
-      setActivities(data);
+      if (!all) {
+        const { data } = await activityService.getActivities(
+          otherUserID ? otherUserID : user.id
+        );
+        setActivities(data);
+      } else {
+        const { data } = await activityService.getAllActivities();
+        setActivities(data);
+      }
```

Comment:
`activityService.getActivities` and `activityService.getAllActivities`

is quite confusing

maybe you can rename it to like below

activityService.getUser
activityService.getAllUsers

No need to mention `activities` again like `getUserActivities` since it is understandable that it is for activities because of `activityService` name